src/fides/api/ops/models/privacy_preference.py

```diff
@@ -0,0 +1,206 @@
+
+        There is only one CurrentPrivacyPreference for each PrivacyNotice/ProvidedIdentity.
+        """
+        privacy_notice_history = PrivacyNoticeHistory.get(
+            db=db, object_id=data.get("privacy_notice_history_id")
+        )
+        if not privacy_notice_history:
+            raise PrivacyNoticeHistoryNotFound()
+
+        data["relevant_systems"] = privacy_notice_history.calculate_relevant_systems(db)
+        created_privacy_preference_history = super().create(
+            db=db, data=data, check_name=check_name
+        )
+
+        current_privacy_preference_data = {
+            "preference": created_privacy_preference_history.preference,
+            "provided_identity_id": created_privacy_preference_history.provided_identity_id,
+            "privacy_notice_id": privacy_notice_history.privacy_notice_id,
+            "privacy_notice_history_id": privacy_notice_history.id,
+            "privacy_preference_history_id": created_privacy_preference_history.id,
+        }
+
+        existing_current_preference = (
+            db.query(CurrentPrivacyPreference)
+            .filter(
+                CurrentPrivacyPreference.provided_identity_id
+                == created_privacy_preference_history.provided_identity_id,
+                CurrentPrivacyPreference.privacy_notice_id
+                == privacy_notice_history.privacy_notice_id,
+            )
+            .first()
+        )
+
+        if existing_current_preference:
+            existing_current_preference.update(
+                db=db, data=current_privacy_preference_data
+            )
+        else:
+            CurrentPrivacyPreference.create(
+                db=db, data=current_privacy_preference_data, check_name=False
+            )
+
+        return created_privacy_preference_history
+
+
+class CurrentPrivacyPreference(Base):
+    """Stores only the user's most recently saved preference for a given privacy notice
+
+    The specific privacy notice history and privacy preference history record are linked as well.
+    """
+
+    preference = Column(EnumColumn(UserConsentPreference), nullable=False, index=True)
+    provided_identity_id = Column(
+        String,
+        ForeignKey(ProvidedIdentity.id),
+        nullable=True,
+    )
+    privacy_notice_id = Column(
+        String, ForeignKey(PrivacyNotice.id), nullable=False, index=True
+    )
+    privacy_notice_history_id = Column(
+        String, ForeignKey(PrivacyNoticeHistory.id), nullable=False, index=True
+    )
+    privacy_preference_history_id = Column(
+        String, ForeignKey(PrivacyPreferenceHistory.id), nullable=False, index=True
+    )
+
+    privacy_notice_history = relationship(PrivacyNoticeHistory)
+    privacy_preference_history = relationship(
+        PrivacyPreferenceHistory, cascade="delete, delete-orphan", single_parent=True
+    )
+
+    UniqueConstraint(
+        provided_identity_id, privacy_notice_id, name="identity_privacy_notice"
+    )
```

Comment:
The main purpose of this table is to store the user's Current Privacy Preferences - it stores the latest preference saved for a unique privacy notice, not privacy notice history, although we store the history here too so it's clear which one they consented to. This table is analogous to the `Consent` table and was created to avoid messy queries on the bulky `PrivacyPreferenceHistory` table just to get current preferences.